src/matrixvariates.jl

```diff
@@ -22,16 +22,42 @@ Return the mean matrix of `d`.
 """
 mean(d::MatrixDistribution)
 
-# sampling
+## sampling
+# multiple matrix-variates with pre-allocated array
+function rand(rng::AbstractRNG, s::Sampleable{Matrixvariate},
```

Comment:
Seouls this be a bang function?

Reply:
The problem is that I found three cases for matrix-variates - 1. Nothing is pre-allocated, just rand() 2. The matrices themselves are allocated, so almost no allocation is needed, definitely rand!() 3. The array to stick the matrices in is allocated, but not the matrices themselves - personally I wouldn’t even offer this option, but it was already there, and I couldn’t see how to dispatch differently on it, so I was left with calling it rand(). It was the only really nasty compromise I found to maintain backward compatibility... I’d prefer to drop it?